.github/workflows/windows.yml

```diff
@@ -0,0 +1,142 @@
+    name: "MSVC 14 Release (win-2019)"
+    runs-on: windows-2019
+    steps:
+      - uses: actions/checkout@v1
+      - name: "Log environment"
+        shell: pwsh
+        run: .\scripts\log-env.ps1
+      - name: "Install packages"
+        shell: pwsh
+        run: |
+          vcpkg install libpng sdl1 sdl1-net
+          vcpkg integrate install
+      - name: "Build"
+        shell: pwsh
+        env:
+          PATH: '${env:PATH};C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\amd64'
+        run: |
+          # build steps
+          cd visualc_net
+          MSBuild -m dosbox.sln -p:Configuration=Release
+
+
+  build_msys2_gcc_64:
+    name: MSYS2 GCC-9 x86_64
+    runs-on: windows-2019
+    env:
+      CHERE_INVOKING: yes
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install msys2 environment
+        run:  choco install msys2 --no-progress
+
+      - name: "Log environment"
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/log-env.sh"
+
+      - name: Install dependencies
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/list-build-dependencies.sh | xargs pacman -S --noconfirm"
+
+      - name: Build
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/build.sh --bin-path /mingw64/bin"
+
+      - name: Summarize warnings
+        run:  C:\tools\msys64\usr\bin\python3 ./scripts/count-warnings.py build.log
+
+
+  build_msys2_gcc_32:
+    name: MSYS2 GCC-9 i686
+    runs-on: windows-2019
+    env:
+      CHERE_INVOKING: yes
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install msys2 environment
+        run:  choco install msys2 --no-progress
+
+      - name: "Log environment"
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/log-env.sh"
+
+      - name: Install dependencies
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/list-build-dependencies.sh --bit-depth 32 | xargs pacman -S --noconfirm"
+
+      - name: Build
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/build.sh --bit-depth 32 --bin-path /mingw32/bin"
+
+      - name: Summarize warnings
+        run:  C:\tools\msys64\usr\bin\python3 ./scripts/count-warnings.py build.log
+
+
+  build_msys2_clang_64:
+    name: MSYS2 Clang-8 x86_64
+    runs-on: windows-2019
+    env:
+      CHERE_INVOKING: yes
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install msys2 environment
+        run:  choco install msys2 --no-progress
+
+      - name: "Log environment"
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/log-env.sh"
+
+      - name: Install dependencies
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/list-build-dependencies.sh --compiler clang | xargs pacman -S --noconfirm"
+
+      - name: Build
+        run:  C:\tools\msys64\usr\bin\bash -lc "./scripts/build.sh --compiler clang --bin-path /mingw64/bin"
+
+      - name: Summarize warnings
+        run:  C:\tools\msys64\usr\bin\python3 ./scripts/count-warnings.py build.log
```

Comment:
I still think we don't need to route this step via python3 inside msys, but it's not a blocker.

Reply:
I don't know why python3 isn't showing up in the PATH; I didn't look - maybe your original build.sh augmented PATH in the YAML (which I don't have)?

Reply:
Ah, my bad, I looked at results of wrong job in my test commit 65be9fc4da348a021fedb

Following line will work:

      - name: Summarize warnings
        run: .\scripts\count-warnings.py build.log

I think at some point in recent past the default shell on Windows CI machines changed (at least I remember reading in the documentation it was `cmd` and now it says `powershell` is the default).

Reply:
Will give it a whirl!  

Great to see powershell (and Windows in general) being able to launch scripts directly now (does it actually parse the first #! line or does it look at the .py extension?).

Reply:
Confirmed working!
All builds are passing now; although details in the macOS clang build appears it didn't make it to the count-warnings step - so I'm running it again.